contracts/exchangeIssuance/ExchangeIssuanceLeveragedForCompound.sol

```diff
@@ -0,0 +1,1346 @@
+     */
+    function _swapCollateralForOutputToken(
+        address _collateralToken,
+        uint256 _collateralTokenAmount,
+        address _outputToken,
+        uint256 _minAmountOutputToken,
+        DEXAdapter.SwapData memory _swapData
+    )
+    internal
+    isValidPath(_swapData.path, _collateralToken, _outputToken)
+    returns (uint256)
+    {
+        return addresses.swapExactTokensForTokens(
+            _collateralTokenAmount,
+            _minAmountOutputToken,
+            _swapData
+        );
+    }
+
+    /**
+     * Deposit collateral to compound to obtain collateralCToken for issuance
+     *
+     * @param _collateralToken              Address of collateral token
+     * @param _depositAmount                Amount to deposit
+     */
+    function _depositToCompound(
+        address _collateralToken,
+        uint256 _depositAmount
+    ) internal {
+        // get cTokenAddress from underlying assets
+        address cTokenAddress = _getCTokenAddess(_collateralToken);
+
+        if (_collateralToken != addresses.weth) {
+            IERC20(_collateralToken).approve(cTokenAddress, _depositAmount);
+            ICErc20Delegator(cTokenAddress).mint(_depositAmount);
+        } else {
+            IWETH(addresses.weth).withdraw(_depositAmount);
+            ICEther(payable(cTokenAddress)).mint{value: _depositAmount}();
+            // return mintTokens;
+        }
+    }
+
+    /**
+     * Convert collateralCToken from set redemption to collateralToken by withdrawing underlying from Compound
+     *
+     * @param _collateralToken       Address of the collateralToken to withdraw from Compound
+     * @param _collateralAmount      Amount of collateralToken to withdraw
+     */
+    function _withdrawFromCompound(
+        address _collateralToken,
+        uint256 _collateralAmount
+    ) internal returns (uint256){
+        uint256 result;
+
+        // get cTokenAddress from underlying assets
+        address cTokenAddress = _getCTokenAddess(_collateralToken);
+        
+        Exp memory exchangeRate = Exp({mantissa: ICEther(payable(cTokenAddress)).exchangeRateStored()});
+        uint256 cTokenAmount = div_((_collateralAmount), exchangeRate);
+
+        uint256 balance = ICErc20Delegator(cTokenAddress).balanceOf(address(this));
+        require(balance >= cTokenAmount, "ExchangeIssuance: Collateral Amount issue");
+        
+
+        if (_collateralToken != addresses.weth) {
+            result = ICErc20Delegator(cTokenAddress).redeem(cTokenAmount);
+        } else {
+            result = ICEther(payable(cTokenAddress)).redeem(cTokenAmount);
+            
+            IWETH(addresses.weth).deposit{value: (_collateralAmount - ROUNDING_ERROR_MARGIN)}();
+        }
+        return result;
+    }
+
+    /**
+     * Returns the cToken Address from underlying asset Address
+     *
+     * @param _collateralToken       Address of the the collateral token
+     */
+    function _getCTokenAddess(
```

Comment:
Once you have replaced the usages of this function with just using the existing data, you can remove this function, as well as the `CompoundLeverageModuleStorage` interface

Reply:
addressed in 2bc43dfd123d9c826c22e1931048af595bb3ebc3